R/dbpath.R

```diff
@@ -80,14 +80,31 @@ setMethod("dbConnect", "dbpath", function(drv) {
 #' @export
 print.dbpath <- function(x, hide_password = TRUE, ...) {
   # name, username, password, ipv4host, port, database
-  print(
-    paste0(
-      x[["name"]], "://",
-      x[["username"]], ":", if (hide_password) "****" else x["password"], "@",
-      x[["host"]],
-      if (x[["port"]] != "") ":" else "", x["port"],
-      if (x[["database"]] != "") "/" else "", x["database"]
-    )
+  cat("<dbpath>\n", format(x, hide_password = hide_password, ...), sep = "")
+}
+
+#' Format a dbpath object
+#'
+#' Returns a formatted dbpath URL as a character string.
+#'
+#' @param x The [dbpath()] object
+#' @inheritParams print.dbpath
+#'
+#' @return A character string consisting of a dbpath URL, e.g
+#'   `<dialect>+<driver>://<username>:<password>@<host>:<port>/<database>`.
+#'
+#' @export
+format.dbpath <- function(x, hide_password = FALSE, ...) {
```

Comment:
Printing the dbpath hides the password (since it's most likely interactive), but formatting it into the URL shows the password (since you're more likely to be doing something programmatic).